server/src/main/java/org/elasticsearch/common/logging/ESLoggerFactory.java

```diff
@@ -46,6 +46,9 @@ public static Logger getLogger(String prefix, Class<?> clazz) {
     }
 
     public static Logger getLogger(String prefix, Logger logger) {
+        if (prefix == null || prefix.length() == 0) {
```

Comment:
Most of our loggers don't actually have a prefix so I felt like we shouldn't wrap the default logger in that case. It is one less object to allocate and one less synchronized block to execute.